File: gestionApp/Evento/Evento.java
package gestionApp.Evento;
import gestionApp.Negocio;

import java.io.Serializable;
import java.time.DayOfWeek;
import java.time.LocalDateTime;
import java.util.ArrayList;



public class Evento implements Tiempo, Serializable {

    protected String id;
    protected Negocio negocio;
    protected String descripcion;
    protected LocalDateTime fecha;
    protected String respuesta;
    private static int idCounter = 0;
    private static ArrayList<Evento> allEventos = new ArrayList<>();
    private static ArrayList<Evento> allEventosPendientes = new ArrayList<>();


    public Evento(Negocio negocio, String descripcion, LocalDateTime fecha){   
            
        this.id = "0".repeat(3-String.valueOf(idCounter).length()) + String.valueOf(idCounter);
        this.negocio = negocio;
        this.descripcion = descripcion;
        this.fecha = fecha;
        if (this.negocio.getEtapa().equals("Finalizado")){
            System.out.println("No se permiten crear mas Eventos para este negocio, ya que fue cerrado anteriormente");
        }
        else{
            this.negocio.agregarEvento(this);
            Evento.allEventos.add(this);
        }
        idCounter += 1;
        
    }

    //---------------Metodos Setter y Getter---------------

    protected String getId() {return id;}

    public Negocio getNegocio() {return negocio;}
    public void setNegocio(Negocio negocio) {this.negocio = negocio;}

    public String getDescripcion() {return descripcion;}
    public void setDescripcion(String descripcion) {this.descripcion = descripcion;}

    public LocalDateTime getFecha() {return fecha;}
    public void setFecha(LocalDateTime fecha) {this.fecha = fecha;}

    public String getRespuesta() {return respuesta;}

    public static ArrayList<Evento> getAllEventos() {return allEventos;}

    //Metodo para elegir si una respuesta cambia la etapa del negocio

    public static void cursoEvento(){
    	
    	
    	
        
    }
    
    ///AGENDA//
    

    /*public static void agendarEvento(Llamada llamada) {
    	
    	
    }
   public static void agendarEvento(Correo correo) {
    	
    	
    }
   public static void agendarEvento(Reunion reunion) {
   	
   	
   }
    
    
    
    public static void Agenda(){
    	for(int i = 0; i < allEventos.size(); i++) {
    		
    		DayOfWeek  dia=LocalDateTime.parse((CharSequence) allEventos.get(i)).getDayOfWeek();
    		System.out.println("dia");
    		allEventos.get(i).toString();
    		
    		
    		
    		
    	}
    	
    	
        
    }
    */
    


}

File: Python/gestionApp/eventos/Evento.py
from distutils.dep_util import newer_group
from random import randrange


class Evento:
    allEventos = []
    allEventosPendientes = []
    def __init__(self, negocio, descripcion, fecha):
        self.id = Evento.crear_id()
        self.negocio = negocio
        self.descripcion = descripcion
        self.fecha = fecha
        if (negocio.getEtapa().equals("Finalizado")):
            print("No se permiten crear mas Eventos para este negocio, ya que fue cerrado anteriormente")
        else:
            self.negocio.agregarEvento(self)
            Evento.allEventos.append(self)
            
    def getNegocio(self): return self.negocio
    def setNegocio(self,negocio): self.negocio = negocio
    
    def getDescripcion(self): return self.descripcion
    def setDescripcion(self,description): self.descripcion = description
    
    def getFecha(self): return self.fecha
    def setFecha(self,fecha): self.fecha = fecha
    
    @classmethod
    def getAllEventos(cls): return cls.allEventos
    def setAllEventos(cls): cls.allEventos
    
    def getAllEventosPendientes(cls): return cls.allEventosPendientes
    def setAllEventosPendientes(cls,allEventosPendientes): cls.allEventosPendientes

    @classmethod
    def busquedaEvento(cls, id):
        for i in range(len(Evento.getAllEventos())):
            if Evento.getAllEventos()[i].id == id:
                return Evento.getAllEventos()[i]

    @classmethod
    def eliminarServicio(cls, id):
        for i in range(len(Evento.getAllEventos())):
            if Evento.getAllEventos()[i].id == id:
                Evento.getAllEventos().pop(i)

    @classmethod
    def crear_id(cls):
        idGenerado = randrange(0, 1000, 1)
        if len(Evento.getAllEventos()) == 0:
            return idGenerado
        else:
            for i in range(len(Evento.getAllEventos())):
                if idGenerado == Evento.getAllEventos()[i].id:
                    i = 0
                    idGenerado = randrange(0, 1000, 1)
                elif i == len(Evento.getAllEventos()) - 1:
                    return idGenerado